--- a/src/ReportCSV.c	2007-08-29 23:57:27.000000000 +0200
+++ b/src/ReportCSV.c	2012-10-17 15:54:46.516986280 +0200
@@ -149,9 +149,8 @@
 }
 
 void CSV_timestamp( char *timestamp, int length ) {
-    time_t times;
-    struct tm *timest;
-    times = time( NULL );
-    timest = localtime( &times );
-    strftime( timestamp, length,"%Y%m%d%H%M%S", timest );
+    struct timeval tv;
+    gettimeofday(&tv, NULL);
+    snprintf(timestamp, length, "%lu.%06lu",
+             (unsigned long)tv.tv_sec, (unsigned long)tv.tv_usec);
 }
